layouts/partials/fragments/list.html

```diff
@@ -0,0 +1,108 @@
+{{/*
+  This variable stores all the page bundles in the blog section. There are three
+  where functions achieving this. First where gets all the pages in blog
+  section, the second one removes current page from the mix and the last one
+  removes the section (.Site.Pages returns the current section as a page). */}}
+{{- $articles := where (where .Site.Pages "Section" (.Params.section | default .page.Section)) "Params.fragment" "in" (slice "content-single" "content-split") -}}
```

Comment:
That's is quite a great query. Wouldn't have thought of that.
It implicitly defaults to .page.Section. Does that always work? Or might there be some strange edge cases?

Reply:
Nothing that I'm aware of. The only thing is if the list fragment is used in the root of content directory or any first level directories there wouldn't be any sections. It's just the Hugo's definition of section.